tests/api_connexion/endpoints/test_dag_run_endpoint.py

```diff
@@ -346,6 +346,55 @@ def test_end_date_gte_lte(self, url, expected_dag_run_ids, session):
         self.assertEqual(dag_run_ids, expected_dag_run_ids)
 
 
+class TestGetDagRunBatch(TestDagRunEndpoint):
+    @provide_session
+    def test_should_respond_200(self, session):
+        dag_runs = self._create_test_dag_run()
+        session.add_all(dag_runs)
+        session.commit()
+        payload = {
+            "page_offset": 0,
+            "page_limit": 10,
+            "dag_ids": ["TEST_DAG_I"],
+            "execution_date_gte": "2020-06-24T19:54:56Z",
+            "execution_date_lte": self.default_time,
+            "start_date_gte": "2020-06-24T19:54:56Z",
+            "start_date_lte": self.default_time,
+            "end_date_gte": "2020-06-24T19:54:56Z",
+            "end_date_lte": self.default_time
+        }
+        response = self.client.post("api/v1/dags/~/dagRuns/list", json=payload)
+        assert response.status_code == 200
+        self.assertEqual(
+            response.json,
+            {
+                "dag_runs": [
+                    {
+                        'dag_id': 'TEST_DAG_ID',
+                        'dag_run_id': 'TEST_DAG_RUN_ID_1',
+                        'end_date': None,
+                        'state': 'running',
+                        'execution_date': self.default_time,
+                        'external_trigger': True,
+                        'start_date': self.default_time,
+                        'conf': {},
+                    },
+                    {
+                        'dag_id': 'TEST_DAG_ID',
+                        'dag_run_id': 'TEST_DAG_RUN_ID_2',
+                        'end_date': None,
+                        'state': 'running',
+                        'execution_date': self.default_time_2,
+                        'external_trigger': True,
+                        'start_date': self.default_time,
+                        'conf': {},
+                    },
+                ],
+                "total_entries": 2,
+            }
+        )
+
+
```

Comment:
I would suggest you add more tests. Like testing the page limit and offset, the execution date etc. You may use parameterized for the tests

Reply:
Thanks! @ephraimbuddy .  I will add more test soon.

@ephraimbuddy @mik-laj Could you help me figure out why my test case is failing ? 

I am getting this error:
**TypeError: get_dag_runs_batch() missing 1 required positional argument: 'dag_ids'**

More details is here: https://github.com/apache/airflow/pull/9556/checks?check_run_id=816283026

I get the same error in local dev env. 

Thanks!

Reply:
Parameters for the view function are only passed if the parameters are from a URL  In this case, the parameters are passed as the body, so you need to handle them in the view.
https://connexion.readthedocs.io/en/latest/request.html#automatic-parameter-handling

Reply:
Thanks @mik-laj